geo/src/algorithm/densify_geodesic.rs

```diff
@@ -0,0 +1,243 @@
+use num_traits::{FromPrimitive, ToPrimitive};
+
+use crate::{CoordFloat, CoordsIter, Line, LineString, MultiLineString, MultiPolygon, Point, Polygon, Rect, Triangle};
+
+use crate::{GeodesicIntermediate, GeodesicLength};
+
+/// Returns a new geometry on a spheroid containing both existing and new interpolated coordinates with
+/// a maximum distance of `max_distance` between them.
+///
+/// Note: `max_distance` must be greater than 0.
+///
+/// ## Units
+///
+/// `max_distance`: meters
+///
+/// # Examples
+/// ```
+/// use approx::assert_relative_eq;
+///
+/// use geo::{coord, GeodesicLength, Line, LineString};
+/// use geo::DensifyGeodesic;
+///
+/// let line = Line::new(coord! {x: 10.0, y: 20.0}, coord! { x: 125.0, y: 25.00 });
+/// // known output
+/// let output: LineString = vec![[10.0, 20.0], [65.879360, 37.722253], [125.0, 25.00]].into();
+/// // densify
+/// let dense = line.densify_geodesic(5703861.471800622);
+///
+/// assert_relative_eq!(dense, output, epsilon = 1.0e-6);
+///```
+pub trait DensifyGeodesic<F: CoordFloat> {
+    type Output;
+
+    fn densify_geodesic(&self, max_distance: F) -> Self::Output;
```

Comment:
```suggestion
    fn densify_geodesic(&self, max_meters: f64) -> Self::Output {
```

I know it's in the docs, but it's nice to be redundant sometimes.